habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -261,39 +275,91 @@ def reconfigure(self, config: Config):
                 ao: ao.joint_positions for ao in self.art_objs
             }
 
+    def _setup_targets(self):
+        self._targets = {}
+        for target_handle, transform in self.ep_info["targets"].items():
+            self._targets[target_handle] = mn.Matrix4(
+                [[transform[j][i] for j in range(4)] for i in range(4)]
+            )
+
+    def get_nav_pos(self, pos):
+        pos = mn.Vector3(*pos)
+        height_thresh = 0.15
+        z_min = -0.2
+        use_vs = np.array(self.pathfinder.build_navmesh_vertices())
+
+        if height_thresh is not None:
+            use_vs = use_vs[use_vs[:, 1] < height_thresh]
+        if z_min is not None:
+            use_vs = use_vs[use_vs[:, 2] > z_min]
+        dists = np.linalg.norm(
+            use_vs[:, [0, 2]] - np.array(pos)[[0, 2]], axis=-1
+        )
+
+        closest_idx = np.argmin(dists)
+        return use_vs[closest_idx]
+
     def _recompute_navmesh(self):
-        """Generates the navmesh on the fly. This must be called
+        """Generates the navmesh or loads the saved navmesh if it exists. This must be called
         AFTER adding articulated objects to the scene.
         """
 
-        # cache current motiontype and set to STATIC for inclusion in the NavMesh computation
-        motion_types = []
-        for art_obj in self.art_objs:
-            motion_types.append(art_obj.motion_type)
-            art_obj.motion_type = MotionType.STATIC
-        # compute new NavMesh
-        self.recompute_navmesh(
-            self.pathfinder,
-            self.navmesh_settings,
-            include_static_objects=True,
-        )
-        # optionally save the new NavMesh
-        if self.habitat_config.get("SAVE_NAVMESH", False):
-            scene_name = self.ep_info["scene_id"]
-            inferred_path = scene_name.split(".glb")[0] + ".navmesh"
-            self.pathfinder.save_nav_mesh(inferred_path)
-        # reset cached MotionTypes
-        for art_obj, motion_type in zip(self.art_objs, motion_types):
-            art_obj.motion_type = motion_type
+        scene_name = self.ep_info["scene_id"]
+        navmesh_path = scene_name.split(".glb")[0] + ".navmesh"
```

Comment:
This is looking for a stage navmesh in the same location with the same name. This is generally not the case for our datasets anymore. Instead, the scene may come with a NavMesh in config in which case it will be loaded automatically when the scene is initialized. Generally, if the NavMesh is not loaded during initialization it should be recomputed.

Reply:
The navmesh may need to be recomputed if the state of articulated objects changes. Therefore, I believe it's important to keep the code that loads and saves the navigation mesh, the defaults won't work a lot of the time. I will have to wait until the navigation mesh is updated in ReplicaCAD v1.4 to utilize the ones included with the dataset.